src/components/modals/SelectSpeakerModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState, useLayoutEffect } from 'react';
+import {
+    Text,
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalBody,
+    ModalFooter,
+    Button,
+    Checkbox,
+    Box
+} from "@chakra-ui/react";
+import { OnlySelectField } from '../sideBarContentEdit/SelectField';
+import { useGetMetadata } from '@/services/api/transcripts/useGetMetadata';
+
+type Props = {
+    selectedSpeakerElement: any;
+    isOpen: boolean;
+    onSetSpeakerName: (newSpeakerName: string | null, isUpdateAllSpeakerInstances: boolean) => void;
+    onClose: () => void;
+};
+
+const SelectSpeakerModal = ({ selectedSpeakerElement, onSetSpeakerName, onClose: closeModal }: Props) => {
+    const oldSpeakerName = selectedSpeakerElement?.speaker || ''
+    const [newSpeakerName, setNewSpeakerName] = useState(oldSpeakerName);
+    const [isUpdateAllSpeakerInstances, setIsUpdateAllSpeakerInstances] = useState(false);
+    const { data: allMetadata } = useGetMetadata();
+
+
+    useLayoutEffect(() => {
```

Comment:
Yes, there was an inconsistency where a previous version of selected speaker was shown in the modal selection, and because `useLayoutEffect` "fires before the browser repaints the screen" ([src](https://react.dev/reference/react/useLayoutEffect)), that issues was solved by using it.

I just read in the linked docs that it can hurt performance. I don't recall of putting a lot of effort to see if this issue was rooted somewhere else, thus avoiding `useLayoutEffect`. I'll check it out.

Reply:
Good catch, I improved the state handling logic and `useLayoutEffect` is not needed anymore.